src/main/scala/com/trueaccord/scalapb/json/JsonFormat.scala

```diff
@@ -358,7 +358,7 @@ object JsonFormat {
     case (ScalaType.Int, JNull) => PInt(0)
     case (ScalaType.Long, JLong(x)) => PLong(x.toLong)
     case (ScalaType.Long, JDecimal(x)) => PLong(x.longValue())
-    case (ScalaType.Long, JString(x)) => PLong(x.toLong)
+    case (ScalaType.Long, JString(x)) => PLong(java.lang.Long.parseUnsignedLong(x))
```

Comment:
This would be incorrect. Longs can be int64 which are allowed to be negative.  Can you mimic the behaviour at https://github.com/google/protobuf/blob/master/java/util/src/main/java/com/google/protobuf/util/JsonFormat.java ?